### PR TITLE
Add maximum publication rate parameter

### PR DESCRIPTION
**Pull request by dyackzan**
This will expose an interface by which we can limit the rate of published images and reduce the load on the PC running this package.

### CODE REVIEW

File: src/vision.cpp [hunk 1]
```diff
@@ -124,6 +125,19 @@ bool Vision::configure()
...
+  node_->declare_parameter<double>(MAX_PUB_RATE_PARAM);
+  if (!node_->get_parameter<double>(MAX_PUB_RATE_PARAM, max_pub_rate_hz_))
+  {
+    max_pub_rate_hz_ = 30.0;
+    RCLCPP_WARN(node_->get_logger(), "No maximum publication rate set, using max_pub_rate '%f'", max_pub_rate_hz_);
+    node_->set_parameter(rclcpp::Parameter(MAX_PUB_RATE_PARAM, max_pub_rate_hz_));
+  }
+  else
+  {
+    RCLCPP_WARN(node_->get_logger(), "Should have successfully set max_pub_rate: '%f'", max_pub_rate_hz_);
```
Comment (eholum) [inline src/vision.cpp]:
```suggestion
    RCLCPP_WARN(node_->get_logger(), "Using max_pub_rate: '%f'", max_pub_rate_hz_);
```

File: src/vision.cpp [hunk 2]
```diff
@@ -549,6 +563,7 @@ void Vision::run()
       }
     }
     rclcpp::spin_some(node_);
+    max_pub_rate_->sleep();
```
Comment (eholum) [inline src/vision.cpp:566]: Is anything being caught in the sink or is it always pulling the latest image? ie, if you set the rate to something very very slow is it up to date our do we see lag?

File: src/vision.cpp [hunk 3]
```diff
@@ -124,6 +125,19 @@ bool Vision::configure()
     node_->set_parameter(rclcpp::Parameter(FRAME_ID_PARAM, frame_id_));
   }
 
+  node_->declare_parameter<double>(MAX_PUB_RATE_PARAM);
+  if (!node_->get_parameter<double>(MAX_PUB_RATE_PARAM, max_pub_rate_hz_))
+  {
+    max_pub_rate_hz_ = 30.0;
```
Comment (eholum) [inline src/vision.cpp:131]: What happens if you set this to a higher publisher rate than the camera is streaming at? iirc it maxes out at 30 hz, but can be as low as 6 for depth images? Does it matter?

**Review by eholum — COMMENTED** [review]
Mostly just some testing edge case questions. If this all works then I think it's fine for our purposes in the near future! Someday we should refactor the publisher and spin functions, but can handle that down the road.

File: src/vision.cpp (hunk 3 above)
Comment (dyackzan) [reply src/vision.cpp:131]: Yeah it just looks like `rate.sleep()` doesn't make any difference to the frequency of publication when it's set to a higher frequency than the camera streams at

File: src/vision.cpp (hunk 2 above)
Comment (dyackzan) [reply src/vision.cpp:566]: Looks like it pulls the latest image. I set the rate to 0.5 and I still see recent data

File: src/vision.cpp (hunk 1 above)
Comment (dyackzan) [reply src/vision.cpp]: Oh, good catch. Had this in here for debugging

**Review by eholum — APPROVED** [review]

**Pull request merged**